src/Features/CSharp/Portable/AddNonNullTypes/CSharpAddNonNullTypesCodeFixProvider.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.AddNonNullTypes
+{
+    /// <summary>
+    /// When some nullable-related syntax is encountered outside of a NonNullTYpes context,
+    /// offer to add a type-level NonNullTypes attribute.
+    /// </summary>
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.AddNonNullTypes), Shared]
+#pragma warning disable RS1016 // This provider intentionally does not provide a FixAll
```

Comment:
why not?

Reply:
I don't think that's very useful. Most likely you should use a module-level attribute instead for such cases.
Also, I suspect it's not trivial to implement properly.
The cost/benefit combination makes it low priority at the moment (we're trying to address blockers before preview 1).

Reply:
I'll leave it to you to decide the right design.  However:

> Also, I suspect it's not trivial to implement properly.

This should be very trivial.  Basically, you derive from SyntaxEditorBasedCodeFixProvider, and implement FixAll.  Your fixall will wlak all your diagnostics and get the unique set of TypeDecls to fix.  Then it will just walk all of those and add attributes like you do today.

Reply:
Yes, that'd be the core, but what about FixAll across documents (partial type scenarios)?
`SyntaxEditorBasedCodeFixProvider` seems to work on a document basis.

Reply:
Or just fix this part: 

>  (the attribute can't be specified multiple times).

I personally get super frustrated with these types of attributes because now i have to deal with that :)

Reply:
>  the scenario is that you have a partial class where the 'first part' doesn't involve any nullable notations (thus no warnings), and some other part does.

Gotcha.  An interesting case to be sure.  It does feel like we could expose a more flexibly entrypoint in SyntaxEditorBasedCodeFixProvider.  One where you got to handle all the diagnostics at once, instead of us handling each document one at a time, having you only fixup that document.